scripts/check_modules.py

```diff
@@ -0,0 +1,59 @@
+# This script checks module.json for:
+#
+# 1. all commits references exist in the linear history of the current branch
+# 2. there are no duplicate numeric versions for the same module ID, say `go-1.19-v1-20230412-9uegy7e` 
+#    and `go-1.19-v1-20230501-cceo023`
+
+import subprocess
+import os
+import json
+
+module_registry_file = 'modules.json'
+
+def commit_exists(commit):
+  try:
+    output = subprocess.check_output(['git', 'show', '-s', '--format=format:%H', commit])
+    return str(output, 'UTF-8') == commit
+  except:
+    return False
+
+def get_commits():
+  if not os.path.isfile(module_registry_file):
+    return {}
+  f = open(module_registry_file, 'r')
+  modules = json.load(f)
+  f.close()
+  commits = set()
+  for entry in modules.values():
+    commits.add(entry['commit'])
+  return commits
+
+def check_commits_exist():
+  for commit in get_commits():
+    assert commit_exists(commit), 'Commit %s does not exist!' % commit
+
+def get_modules():
+  if not os.path.isfile(module_registry_file):
+    return {}
+  f = open(module_registry_file, 'r')
+  modules = json.load(f)
+  f.close()
```

Comment:
nit: should probably open the file once then use that in both checks. otherwise once `modules.json` gets large (and it will get large) this script will take *forever* and potentially OOM (depending on python GC)